portfolio/src/main/webapp/script.js

```diff
@@ -40,14 +40,28 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-function getData() {
-  fetch('/data').then(response => response.json()).then(arr => {
-    const dataContainer = document.getElementById('data-container');
-    arr.forEach(str => {
-      var comment = document.createElement('p');
-      comment.innerText = str;
-      dataContainer.appendChild(document.createElement('hr'));
-      dataContainer.appendChild(comment);
+function initPage() {
+    fetch('/data').then(response => response.json()).then(arr => {
+        const dataContainer = document.getElementById('data-container');
+        arr.forEach(str => {
+            var comment = document.createElement('p');
+            comment.innerText = str;
+            dataContainer.appendChild(document.createElement('hr'));
+            dataContainer.appendChild(comment);
+        });
     });
-  });
-}
\ No newline at end of file
+    fetch('/user').then(response => response.json()).then(userStatus => {
+        const userContainer = document.getElementById('user-container');
+        if (userStatus.isUserLoggedIn) {
+            userContainer.innerHTML = '<span style="float:left;">username: ' + userStatus.userEmail + '</span>'
+                + '<span style="float:right;"><a href="' + userStatus.logoutUrl + '">logout</a></span>';
+        }
+        else {
+            userContainer.innerHTML = '<span style="float:left;">username: None</span>'
+                + '<span style="float:right;"><a href="' + userStatus.loginUrl + '">login</a></span>';
+        }
```

Comment:
just question so doesn't need to modify: I guess cloud shell doesn't provide any hint or suggestion about line length?